include/small/ibuf.h

```diff
@@ -160,7 +179,9 @@ static inline void
 ibuf_truncate(struct ibuf *ibuf, size_t used)
 {
 	assert(used <= ibuf_used(ibuf));
+	size_t size = ibuf->wpos - (ibuf->rpos + used);
 	ibuf->wpos = ibuf->rpos + used;
+	ASAN_POISON_MEMORY_REGION(ibuf->wpos, size);
```

Comment:
Could just poison `ibuf->end - ibuf->wpos`, like you do in `ibuf_alloc`. Actually, you always poison or unpoison the same range so it's probably worth adding some helper functions for that.

Reply:
Introduced `ibuf_poison_unallocated`/`ibuf_unpoison_unallocated`.

<details>
<summary>diff</summary>

```diff
diff --git a/include/small/ibuf.h b/include/small/ibuf.h
index 1e9c848..472c116 100644
--- a/include/small/ibuf.h
+++ b/include/small/ibuf.h
@@ -113,12 +113,24 @@ ibuf_pos(struct ibuf *ibuf)
 	return ibuf->rpos - ibuf->buf;
 }
 
+static inline void
+ibuf_poison_unallocated(const struct ibuf *ibuf)
+{
+	ASAN_POISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+}
+
+static inline void
+ibuf_unpoison_unallocated(const struct ibuf *ibuf)
+{
+	ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+}
+
 /** Forget all cached input. */
 static inline void
 ibuf_reset(struct ibuf *ibuf)
 {
 	ibuf->rpos = ibuf->wpos = ibuf->buf;
-	ASAN_POISON_MEMORY_REGION(ibuf->buf, ibuf->end - ibuf->buf);
+	ibuf_poison_unallocated(ibuf);
 }
 
 void *
@@ -128,7 +140,7 @@ static inline void *
 ibuf_reserve(struct ibuf *ibuf, size_t size)
 {
 	if (ibuf->wpos + size <= ibuf->end) {
-		ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+		ibuf_unpoison_unallocated(ibuf);
 		return ibuf->wpos;
 	}
 	return ibuf_reserve_slow(ibuf, size);
@@ -144,7 +156,7 @@ ibuf_alloc(struct ibuf *ibuf, size_t size)
 		 * allocated memory after previous ibuf_alloc or poison after
 		 * newly allocated memory after previous ibuf_reserve.
 		 */
-		ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+		ibuf_unpoison_unallocated(ibuf);
 		ptr = ibuf->wpos;
 	} else {
 		ptr = ibuf_reserve_slow(ibuf, size);
@@ -152,7 +164,7 @@ ibuf_alloc(struct ibuf *ibuf, size_t size)
 			return NULL;
 	}
 	ibuf->wpos += size;
-	ASAN_POISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+	ibuf_poison_unallocated(ibuf);
 	return ptr;
 }
 
@@ -172,9 +184,8 @@ static inline void
 ibuf_truncate(struct ibuf *ibuf, size_t used)
 {
 	assert(used <= ibuf_used(ibuf));
-	size_t size = ibuf->wpos - (ibuf->rpos + used);
 	ibuf->wpos = ibuf->rpos + used;
-	ASAN_POISON_MEMORY_REGION(ibuf->wpos, size);
+	ibuf_poison_unallocated(ibuf);
 }
 
 static inline void *
diff --git a/small/ibuf.c b/small/ibuf.c
index bfb94bd..30910b3 100644
--- a/small/ibuf.c
+++ b/small/ibuf.c
@@ -79,7 +79,7 @@ ibuf_reserve_slow(struct ibuf *ibuf, size_t size)
 	 * Otherwise, get a bigger buffer.
 	 */
 	if (size + used <= capacity) {
-		ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+		ibuf_unpoison_unallocated(ibuf);
 		memmove(ibuf->buf, ibuf->rpos, used);
 	} else {
 		/* Use start_capacity as allocation factor. */
@@ -138,5 +138,5 @@ ibuf_shrink(struct ibuf *ibuf)
 	ibuf->rpos = ptr;
 	ibuf->wpos = ptr + used;
 	ibuf->end = ptr + slab_capacity(slab);
-	ASAN_POISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
+	ibuf_poison_unallocated(ibuf);
 }

```

</detail>